Resources/doc/versioning.rst

```diff
@@ -0,0 +1,180 @@
+
+    # app/config/config.yml
+    fos_rest:
+        versioning: true
+
+If you do not want to allow all the methods described above, you should choose which version resolver to enable:
+
+.. code-block:: yaml
+configyml
+    fos_rest:
+        versioning:
+            enabled: true
+            resolvers:
+                query: true # Query parameter: /users?version=v1
+                custom_header: true # X-Accept-Version header
+                media_type: # Accept header
+                    enabled: true
+                    regex: '/(v|version)=(?P<version>[0-9\.]+)/'
+
+You can also choose the guessing order:
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    fos_rest:
+        versioning:
+            enabled: true
+            guessing_order:
+                - query
+                - custom_header
+                - media_type
+
+The matched version is set as a Request attribute with the name ``version``,
+and when using JMS serializer it is also set as an exclusion strategy
+automatically in the ``ViewHandler``.
+
+If you use the method using the ``Accept`` header, to make the version mechanism working:
+
+1 - The format listener must be enabled
+
+See :doc:`Format Listener <format_listener>`
+
+2 - The client must pass the requested version in his header like this :
+
+.. code-block:: yaml
+
+    Accept:application/json;version=1.0
+
+3 - You must have declared the version value in your config, otherwise it won't be catched :
+
+.. code-block:: yaml
+
+    fos_rest:
+        view:
+            mime_types:
+                json: ['application/json', 'application/json;version=1.0', 'application/json;version=1.1']
+
+Note: If you have to handle huge versions and mime types, you can simplify the configuration with a php script:
+
+.. code-block:: php
+
+    // app/config/fos_rest_mime_types.php
+    $versions = array(
+        '1.0',
+        '1.1',
+        '2.0',
+    );
+
+    $mimeTypes = array(
+        'json' => array(
+            'application/json',
+        ),
+        'yml'  => array(
+            'application/yaml',
+            'text/yaml',
+        ),
+    );
+
+    array_walk($mimeTypes, function (&$mimeTypes, $format, $versions) {
+        $versionMimeTypes = array();
+        foreach ($mimeTypes as $mimeType) {
+            foreach ($versions as $version) {
+                array_push($versionMimeTypes, sprintf('%s;version=%s', $mimeType, $version));
+                array_push($versionMimeTypes, sprintf('%s;v=%s', $mimeType, $version));
+            }
+        }
+        $mimeTypes = array_merge($mimeTypes, $versionMimeTypes);
+    }, $versions);
+
+    $container->loadFromExtension('fos_rest', array(
+        'view' => array(
+            'mime_types' => $mimeTypes,
+        ),
+    ));
+
+And then, import it from your config.yml file:
+
+.. code-block:: yaml
+
+    imports:
+        - { resource: assets_version.php }
+
+If you have to verify if the version is correctly catched you can use something like :
```

Comment:
the profiler shows the request attributes its quite simple to use .. I would just remove this entire hint